### DISCUSSION TITLE
p.248 リスト20.28のGenerateTokenの呼び出し方

### DISCUSSION THREAD

**Discussion by kdnakt** [budougumi0617/go_todo_app#52] [Ideas]
`l.TokenGenerator.GenerateToken(ctx, *u)`
となっていますが、次のページでのインターフェースの定義は
`GenerateToken(ctx context.Context, u entity.User) ([]byte, error)`
となっており、entity.Userの型があっていないようです。

`l.TokenGenerator.GenerateToken(ctx, u)`
が正しいでしょうか？

**Comment by budougumi0617** [comment]
直前の `u, err := l.Repo.GetUser(ctx, l.DB, name)` で返ってくる `u` 変数は `*entity.User` 型の値です。
https://github.com/budougumi0617/go_todo_app/blob/5146d0b16e4163814d5a6a15e2485bc45ed4abba/store/user.go#L34-L36

よって、 `*u` とすると、  `entity.User` 型の値が手に入ります。
`u` 変数は `*entity.User` 型の値なので、`GenerateToken(ctx context.Context, u entity.User)` の第2引数の `u` （`entity.User`）に代入するには `*u` となります。

**Reply by kdnakt** [reply]
すみません、どこかで実装を間違えていたようで、 `*u` で問題ありませんでした 🙇
